cdap-docs/examples-manual/source/examples/clicks-and-views.rst

```diff
@@ -44,11 +44,18 @@ the *joined* ``PartitionedFileSet``. The ``beforeSubmit`` method prepares the Ma
 two ``Stream``s as input and sets up the ``PartitionedFileSet`` as output, with the appropriate ``PartitionKey``:
 
 .. literalinclude:: /../../../cdap-examples/ClicksAndViews/src/main/java/co/cask/cdap/examples/clicksandviews/ClicksAndViewsMapReduce.java
-:language: java
-     :lines: 58-73
+   :language: java
+   :lines: 58-73
 
 The Mapper class then keys each of the records based upon the ``viewId``. That makes it so that all clicks and views
-for a particular ``viewId`` go to a single Reducer for joining.
+for a particular ``viewId`` go to a single Reducer for joining. In order to do the join properly, the Reducer needs
+to know which source each record is coming from. To make this possible, the Mapper class can get access to the source
```

Comment:
I think this sentence is hard to parse. Better: 
This is possible by calling the ``getInputName`` method of the ``MapReduceTaskContext`` that is passed to ``initialize()``. Note that the Mapper needs to implement ProgramLifeCycle: